### PR TITLE
Cluster metrics aggregation ignores default labels

### PR DESCRIPTION
**Pull request by doochik**
registry.getMetricsAsJSON should merge metric labels and registry default labels

### CODE REVIEW

File: lib/registry.js [hunk 1]
```diff
@@ -82,7 +82,14 @@ class Registry {
 	}
 
 	getMetricsAsJSON() {
-		return this.getMetricsAsArray().map(metric => metric.get());
+		return this.getMetricsAsArray().map(metric => {
+			const item = metric.get();
+			(item.values || []).forEach(val => {
+				// Merge metric labels with registry default labels
+				val.labels = Object.assign({}, this._defaultLabels, val.labels);
```
Comment (SimenB) [inline lib/registry.js]: this mutates, which is not ok

**Comment by SimenB** [comment]
Could you add a test for this?

**Comment by doochik** [comment]
@SimenB add test for `getMetricsAsJSON` and fix mutation

File: lib/registry.js [hunk 2]
```diff
@@ -82,7 +82,21 @@ class Registry {
...
 
 	getMetricsAsJSON() {
-		return this.getMetricsAsArray().map(metric => metric.get());
+		return this.getMetricsAsArray().map(metric => {
+			const item = metric.get();
+			if (!item.values) {
+				return item;
+			}
+
+			item.values = item.values.map(val =>
```
Comment (SimenB) [inline lib/registry.js]: this still mutates `item.values`. Should probably do a `JSON.parse(JSON.stringify(item))` or something

Comment (doochik) [reply lib/registry.js]: metric.get returns new object always, array.map returns new array. So there is no mutation here.

If you insist i'll change this :)

Comment (SimenB) [reply lib/registry.js]: Ah, `get` returns a new object. Then this is fine, thanks!

**Comment by SimenB** [comment]
Could you update the changelog?

**Comment by doochik** [comment]
done

**Comment by siimon** [comment]
lgtm :)

**Comment by doochik** [comment]
Can you release new version with this patch?

**Pull request merged**